jmix-flowui/flowui/src/main/java/io/jmix/flowui/component/UiComponentUtils.java

```diff
@@ -105,13 +105,18 @@ public static Component getComponent(View<?> view, String id) {
      * or an empty {@link Optional}
      */
     public static Optional<Component> findComponent(Component component, String id) {
-        if (component instanceof View<?> view) {
+        BiPredicate<Component, String> idComparator = findFragment(component) == null
+                ? UiComponentUtils::sameId
+                : FragmentUtils::sameId;
+
+        if (idComparator.test(component, id)) {
```

Comment:
This method searches for **inner** components and doesn't declare that it may return parent, i.e. container itself (and should not).